lib/active_merchant/billing/gateways/forte.rb

```diff
@@ -219,8 +219,12 @@ def message_from(response)
         response['response']['response_desc']
       end
 
-      def authorization_from(response)
-        [response.try(:[], 'transaction_id'), response.try(:[], 'response').try(:[], 'authorization_code')].join('#')
+      def authorization_from(response, parameters)
+        if parameters[:action] == 'capture'
+          [response.try(:[], 'transaction_id'), response.try(:[], 'response').try(:[], 'authorization_code'), parameters[:transaction_id], parameters[:authorization_code]].join('#')
```

Comment:
```suggestion
          [response.dig('transaction_id', 'response', 'authorization_code'), parameters[:transaction_id], parameters[:authorization_code]].join('#')
```

Reply:
Err I think this is different. Maybe `response.dig('transaction_id'), response.dig('response'), response.dig('authorization_code'), parameters[:transaction_id], parameters[:authorization_code].join('#')`

Reply:
Whoops, point. Fixed the suggestion. `response.dig('foo')`, `response['foo']`, and `response.try(:[], 'foo')` are all the same, though, unless I'm missing something, so we might as well go for just `response['foo']` in that case.

Reply:
Updated!